app/cdap/components/WranglerGrid/SelectColumnPanel/SelectColumnsList/index.tsx

```diff
@@ -0,0 +1,196 @@
+  display: flex;
+  margin-right: 10px;
+`;
+
+export default function({
+  transformationDataType,
+  selectedColumnsCount,
+  columnsList,
+  setSelectedColumns,
+  dataQuality,
+  transformationName,
+  selectedColumns,
+}: ISelectColumnsListProps) {
+  const [columns, setColumns] = useState<IHeaderNamesList[]>(columnsList);
+  const [isSingleSelection, setIsSingleSelection] = useState<boolean>(true);
+  const ref = useRef(null);
+
+  const columnsAsPerType = getColumnsSupportedType(transformationDataType, columnsList);
+  const filteredColumnsOnType = getFilteredColumn(transformationDataType, columnsList);
+
+  useEffect(() => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = multipleColumnSelected?.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName
+    );
+
+    if (multiSelect.length) {
+      setIsSingleSelection(false);
+    }
+    setColumns(filteredColumnsOnType);
+  }, []);
+
+  const onSingleSelection = (column: IHeaderNamesList) => {
+    setSelectedColumns([column]);
+  };
+
+  const onMultipleSelection = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    column: IHeaderNamesList
+  ) => {
+    if (event.target.checked) {
+      setSelectedColumns((prev) => [...prev, column]);
+    } else {
+      const indexOfUnchecked = selectedColumns.findIndex(
+        (columnDetail) => columnDetail.label === column.label
+      );
+      if (indexOfUnchecked > -1) {
+        setSelectedColumns(() => selectedColumns.filter((_, index) => index !== indexOfUnchecked));
+      }
+    }
+  };
+
+  const handleDisableCheckbox = () => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = multipleColumnSelected.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName && functionDetail.isMoreThanTwo
+    );
+    if (selectedColumns?.length === 0 || selectedColumns?.length < 2) {
+      return false;
+    } else if (multiSelect.length) {
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value) {
+      const columnValue: IHeaderNamesList[] = filteredColumnsOnType.length
+        ? filteredColumnsOnType.filter((columnDetail: IHeaderNamesList) =>
+            columnDetail.label.toLowerCase().includes(event.target.value.toLowerCase())
+          )
+        : [];
+      columnValue?.length ? setColumns(columnValue) : setColumns([]);
+    } else {
+      setColumns(filteredColumnsOnType);
+    }
+  };
+
+  const handleFocus = () => {
+    ref?.current?.focus();
+  };
+
+  return (
+    <SelectColumnWrapper data-testid="select-column-list-parent">
+      <SelectColumnInnerWrapper>
+        <SelectedColumnCountWidget selectedColumnsCount={selectedColumnsCount} />
+        <SelectColumnSearchBox>
+          <SelectColumnSearchInput data-testid="input_id" onChange={handleSearch} ref={ref} />
```

Comment:
`data-testid` should be more specific. Also, please use hyphens for spaces.

Reply:
Done